lti_consumer/plugin/views.py

```diff
@@ -231,3 +240,153 @@ def scores(self, request, *args, **kwargs):
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
+    """
+    Api Endpoints for LTI NPRS Services
+    """
+    serializer_class = LtiNrpsContextMembershipSerializer
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    def get_page_size(self):
+        """
+        Get Page size for pagination
+        """
+        return settings.REST_FRAMEWORK.get('PAGE_SIZE', 10)
+
+    def build_queryset(self, filter_by=None):
+        """
+        Given a course key return a list of enrollments.
+        """
+        # pylint: disable=import-error,import-outside-toplevel,no-name-in-module
+        from common.djangoapps.student.models import CourseAccessRole
```

Comment:
This is a platform import, it needs to be in `compat.py`.

It seems like we're using too many internal platform methods. It might be better to create a python API in the core that retrieves user data from a course (something like `get_enrolled_users(course_key, roles=None)`).

Do you know a good place to put this API in? I haven't looked at the platform code yet.

Reply:
@giovannicimolin, Does ``openedx.core.djangoapps.enrollments.data`` seems a good place to move this function?

Reply:
https://github.com/edx/edx-platform/pull/25843 has been opened to remove these.